src/SerialFirmata.cpp

```diff
@@ -296,27 +310,43 @@ void SerialFirmata::checkSerial()
         continue;
       }
 #endif
-      if (serialPort->available() > 0) {
-        Firmata.write(START_SYSEX);
-        Firmata.write(SERIAL_MESSAGE);
-        Firmata.write(SERIAL_REPLY | portId);
-
-        if (bytesToRead == 0 || (serialPort->available() <= bytesToRead)) {
-          numBytesToRead = serialPort->available();
-        } else {
-          numBytesToRead = bytesToRead;
+      int availableBytes = serialPort->available();
+      if (availableBytes > 0) {
+        bool read = true;
+        
+#if defined(SERIAL_STORE_AND_FORWARD)    
+        // check if reading should be delayed to collect some bytes before forwarding (for baud rates up to 19200)
```

Comment:
Well, it's an integer division. If the baud rate is higher than 50000 the division result is zero milliseconds and then a timed store&forward is no longer possible, because the timer resolution is only milliseconds. Because of ```(currentMillis - lastReceive[portId]) >= maxCharDelay[portId])``` and assuming the Firmata loop requiring more than 1 millisecond (depending on hardware) one can assume that the effect of the store&forward may cease earlier. Maybe "for baud rates significantly below 57600 baud" is a better description.